lib/utils/bitarray.c

```diff
@@ -311,8 +311,8 @@ int sys_bitarray_xor(sys_bitarray_t *dst, sys_bitarray_t *other, size_t num_bits
 	ret = 0;
 
 out:
-	k_spin_unlock(&dst->lock, key_dst);
 	k_spin_unlock(&other->lock, key_other);
+	k_spin_unlock(&dst->lock, key_dst);
```

Comment:
Bingo.  FWIW best to squash this into the original patch.  No need to have the development history of code outside the tree reflected in git.